fluent-syntax/src/ftlstream.js

```diff
@@ -109,14 +109,50 @@ export class FTLParserStream extends ParserStream {
     return ((cc >= 48 && cc <= 57) || cc === 45); // 0-9
   }
 
-  isPeekNextLineComment() {
+  // -1 - any
+  //  0 - comment
+  //  1 - group comment
+  //  2 - resource comment
+  isPeekNextLineComment(level = -1, oldStyle) {
     if (!this.currentPeekIs('\n')) {
       return false;
     }
 
     this.peek();
 
-    if (this.currentPeekIs('/')) {
+    if (oldStyle === false && this.currentPeekIs('#')) {
+      this.peek();
+      if (level === -1 &&
+        (this.currentPeekIs(' ') || this.currentPeekIs('#'))) {
+        this.resetPeek();
+        return true;
+      }
+      if (level === 0 && this.currentPeekIs(' ')) {
```

Comment:
It looks like this could use a `while` loop where you decrement `level` in each pass.